build/patches/Enable-crash-reporting-ui.patch

```diff
@@ -111,6 +331,46 @@ diff --git a/components/crash_strings.grdp b/components/crash_strings.grdp
 +    Save now
    </message>
  </grit-part>
+diff --git a/components/minidump_uploader/android/java/src/org/chromium/components/minidump_uploader/CrashFileManager.java b/components/minidump_uploader/android/java/src/org/chromium/components/minidump_uploader/CrashFileManager.java
+--- a/components/minidump_uploader/android/java/src/org/chromium/components/minidump_uploader/CrashFileManager.java
++++ b/components/minidump_uploader/android/java/src/org/chromium/components/minidump_uploader/CrashFileManager.java
+@@ -107,6 +107,8 @@ public class CrashFileManager {
+ 
+     private static final Pattern TMP_PATTERN = Pattern.compile("\\.tmp\\z");
+ 
++    private static final String UPLOADED_MINIDUMP_ZIP_SUFFIX = ".zip";
++
+     // The maximum number of non-uploaded crashes that may be kept in the crash reports directory.
+     // Chosen to attempt to balance between keeping a generous number of crashes, and not using up
+     // too much filesystem storage space for obsolete crash reports.
+@@ -116,7 +118,7 @@ public class CrashFileManager {
+     // The maximum age, in days, considered acceptable for a crash report. Reports older than this
+     // age will be removed. The constant is chosen to be quite conservative, while still allowing
+     // users to eventually reclaim filesystem storage space from obsolete crash reports.
+-    private static final int MAX_CRASH_REPORT_AGE_IN_DAYS = 30;
++    private static final int MAX_CRASH_REPORT_AGE_IN_DAYS = 5;
+ 
+     // The maximum number of non-uploaded crashes to copy to the crash reports directory. The
+     // difference between this value and MAX_CRASH_REPORTS_TO_KEEP is that TO_KEEP is only checked
+@@ -460,9 +462,6 @@ public class CrashFileManager {
+     }
+ 
+     public void cleanOutAllNonFreshMinidumpFiles() {
+-        for (File f : getAllUploadedFiles()) {
+-            deleteFile(f);
+-        }
+         for (File f : getAllTempFiles()) {
+             deleteFile(f);
+         }
+@@ -586,6 +585,8 @@ public class CrashFileManager {
+                     && !f.getName().contains(UPLOAD_FORCED_MINIDUMP_SUFFIX)) {
+                 continue;
+             }
++            if (f.getName().contains(UPLOADED_MINIDUMP_ZIP_SUFFIX))
```

Comment:
It should not contain, it should end with that.

Reply:
yes, you are right, although it doesn't matter

Reply:
It would matter if you would write an unit test for it.

Reply:
yes, I have to try